src/Integration/Binding/BindingWorkflow.cs

```diff
@@ -1,395 +1,372 @@
-            notifications.ProgressChanged(message, double.NaN);
-        }
-
-        #endregion
-    }
-}
+//-----------------------------------------------------------------------
+// <copyright file="BindingWorkflow.cs" company="SonarSource SA and Microsoft Corporation">
+//   Copyright (c) SonarSource SA and Microsoft Corporation.  All rights reserved.
+//   Licensed under the MIT License. See License.txt in the project root for license information.
+// </copyright>
+//-----------------------------------------------------------------------
+
+using Microsoft.Alm.Authentication;
+using Microsoft.VisualStudio.CodeAnalysis.RuleSets;
+using SonarLint.VisualStudio.Integration.Persistence;
+using SonarLint.VisualStudio.Integration.Progress;
+using SonarLint.VisualStudio.Integration.Resources;
+using SonarLint.VisualStudio.Integration.Service;
+using SonarLint.VisualStudio.Integration.Service.DataModel;
+using SonarLint.VisualStudio.Integration.TeamExplorer;
+using SonarLint.VisualStudio.Progress.Controller;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Globalization;
+using System.IO;
+using System.Linq;
+using System.Threading;
+
+namespace SonarLint.VisualStudio.Integration.Binding
+{
+    /// <summary>
+    /// Workflow execution for the bind command
+    /// </summary>
+    internal class BindingWorkflow
+    {
+        private readonly BindCommand owner;
+        private readonly ProjectInformation project;
+        private readonly IProjectSystemHelper projectSystemHelper;
+        private readonly IBindingOperation solutionBindingOperation;
+        private readonly ISolutionRuleStore solutionRuleStore;
+
+        internal readonly Dictionary<string, RuleSetGroup> LanguageToGroupMapping = new Dictionary<string, RuleSetGroup>
+        {
+            {SonarQubeServiceWrapper.CSharpLanguage, RuleSetGroup.CSharp },
+            {SonarQubeServiceWrapper.VBLanguage, RuleSetGroup.VB }
+        };
+
+        public BindingWorkflow(BindCommand owner, ProjectInformation project)
+            : this(owner, project, null, null, null)
+        {
+
+        }
+
+        internal /*for testing purposes*/ BindingWorkflow(BindCommand owner, ProjectInformation project, IProjectSystemHelper projectSystemHelper, IBindingOperation solutionBindingOperation, ISolutionRuleStore solutionRuleStore)
+        {
+            if (owner == null)
+            {
+                throw new ArgumentNullException(nameof(owner));
+            }
+
+            if (project == null)
+            {
+                throw new ArgumentNullException(nameof(project));
+            }
+
+            this.owner = owner;
+            this.project = project;
+            this.projectSystemHelper = projectSystemHelper ?? new ProjectSystemHelper(this.owner.ServiceProvider);
+
+            Lazy<SolutionBindingOperation> solutionBindingOperationInstance = new Lazy<SolutionBindingOperation>(

```

Comment:
Matthew's suggestion would be better.
Which field are you referring to?
My suggestion is to keep the internal constructor but just make it do the null checks and assign to the fields (all of which are being passed in as parameters anyway). The fields would still be read-only.
The test helper method would contain the conditional logic to provide defaults for instances that aren't supplied, then would call the product code with a complete set of valid instances to use.